src/app/util/Product/Product.js

```diff
@@ -70,11 +72,15 @@ const getIndexedVariants = variants => variants.map(({ product }) => {
  * @param {{ attribute_code: string }[]} options
  * @returns {number}
  */
-export const getVariantIndex = (variants, options) => variants
-    .findIndex(variant => checkEveryOption(variant.attributes, options));
+export const getVariantIndex = (variants, options) => Object.entries(variants)
+    .filter( ([index,variant]) => {
+        return variant != undefined;
+    }).findIndex(([index,variant]) => checkEveryOption(variant.attributes, options));
 
 export const getVariantsIndexes = (variants, options) => Object.entries(variants)
-    .reduce((indexes, [index, variant]) => {
+    .filter( ([index,variant]) => {
+        return variant != undefined;
+    }).reduce((indexes, [index, variant]) => {
         if (checkEveryOption(variant.attributes, options)) indexes.push(+index);
```

Comment:
Please implement the check for empty variant in the reduce function. This would allow for less loops => higher efficiency.